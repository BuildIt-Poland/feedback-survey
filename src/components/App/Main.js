import React from 'react';
import styled from 'styled-components';
import { element } from 'prop-types';
import { Grid, Box } from 'grommet';

import Header from '../Header';
import Footer from '../Footer';

const Main = styled.main`
  flex-grow: 1;
  display: flex;
  justify-content: center;

  div {
    flex-grow: 1;
    align-items: center;
    justify-content: center;
  }
`;

// const Main = ({ children }) => (
//   <Grid
//     rows={['auto', '1fr', 'auto']}
//     columns={['100%']}
//     areas={[
//       { name: 'header', start: [0, 0], end: [0, 0] },
//       { name: 'main', start: [0, 1], end: [0, 1] },
//       { name: 'footer', start: [0, 2], end: [0, 2] }
//     ]}
//   >
//     <Box gridArea="header">
//       <Header />
//     </Box>
//     <Box gridArea="main">{children}</Box>
//     <Box gridArea="footer">
//       <Footer />
//     </Box>
//   </Grid>
// );

// Main.protoTypes = {
//   children: element.isRequired
// };

export default Main;
