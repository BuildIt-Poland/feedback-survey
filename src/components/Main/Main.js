import React from 'react';
import { Button, Form, FormField, RadioButtonGroup, Select, CheckBox, Box } from 'grommet';

import { colorDarkBlue } from '../../styles/variables';

const messages = {
  required: 'To pole jest wymagane',
  invalid: 'Upsss, błąd!'
};
const Main = function() {
  return (
    <Box align="center" justify="center" pad="medium" background={colorDarkBlue} height="100vh">
      <Form onSubmit={({ value }) => console.log('Submit: ', value)} messages={messages}>
        <FormField name="name" label="Name" required={true} />
        <FormField
          label="Continent"
          name="select"
          component={Select}
          options={['North America', 'South America', 'Asia']}
        />
        <FormField name="address" label="Address" />
        <FormField name="acknowledge" component={CheckBox} label="acknowledge" />
        <FormField name="day" component={RadioButtonGroup} options={['Tuesday', 'Friday']} />
        <Button type="submit" label="Submit" primary={true} />
      </Form>
    </Box>
  );
};

export default Main;
