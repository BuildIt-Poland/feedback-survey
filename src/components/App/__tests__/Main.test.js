import React from 'react';
import { create } from 'react-test-renderer';

import Main from '../Main';

jest.mock('react-router-dom', () => ({
  Link: 'Link'
}));

describe('COMPONENT - App Main', () => {
  it('renders correctly', () => {
    const component = create(<Main />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
