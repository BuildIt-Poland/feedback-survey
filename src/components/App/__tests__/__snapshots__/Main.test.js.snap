// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - App Main renders correctly 1`] = `
<main
  className="sc-bdVaJa RdYzA"
/>
`;
